test/livebook/app_test.exs

```diff
@@ -194,6 +194,29 @@ defmodule Livebook.AppTest do
       user = %{Livebook.Users.User.new() | name: "Jake Peralta"}
       session_id = App.get_session_id(app_pid, user: user)
 
+      assert %{sessions: [%{id: ^session_id, started_by: nil}]} = App.get_by_pid(app_pid)
+    end
+
+    test "stores session creator user in multi-session mode with teams hub" do
+      %{id: hub_id} = Livebook.Factory.insert_hub(:team)
+
+      slug = Utils.random_short_id()
+      app_settings = %{Notebook.AppSettings.new() | slug: slug, multi_session: true}
+
+      notebook = %{
+        Notebook.new()
+        | app_settings: app_settings,
+          hub_id: hub_id,
+          teams_enabled: true
+      }
+
+      app_pid = start_app(notebook)
+
+      assert %{sessions: []} = App.get_by_pid(app_pid)
+
+      user = %{Livebook.Users.User.new() | name: "Jake Peralta"}
+      session_id = App.get_session_id(app_pid, user: user)
+
       assert %{sessions: [%{id: ^session_id, started_by: ^user}]} = App.get_by_pid(app_pid)
 
       # Tracks user updates
```

Comment:
I wonder if we should simplify and not track user updates, now that session-based `:creator` is a marginal use case anyway.

Reply:
Currently in the `App` process we store `created_by_id` and subscribe to `User` updates, so if a session-based user changes the name we pick up the change.

I didn't actually do this for the `Session` process (which also stores `created_by`) and I'm thinking we could always just take the snapshot of what the user is and not worry about updates.

Reply:
Agreed. No need to worry about updates.